src/engraving/dom/articulation.cpp

```diff
@@ -662,6 +662,15 @@ bool Articulation::isOnCrossBeamSide() const
     return chord->beam() && (chord->beam()->cross() || chord->staffMove() != 0) && (up() == chord->up());
 }
 
+staff_idx_t Articulation::vStaffIdx() const
+{
+    ChordRest* cr = chordRest();
+    if (!cr) {
+        return mu::nidx;
```

Comment:
Ok, two things :)
1) Here I would return `staffIdx()`, not `mu::nidx`, because the articulation still has a valid staff even though its parent isn't a ChordRest.
2) I did some investigation. @cbjeukendrup is right: the crash happens because the articulation (in this case a string down bow) has a BarLine as parent. This is how it got written into the file 
<img width="280" alt="image" src="https://github.com/musescore/MuseScore/assets/93707756/b5d3ad77-c25b-4511-9367-1be9f9f1da97">

I found that this can happen if you drag&drop an articulation onto a barline (or select the barline and click the articulation from the palette). This operation shouldn't be allowed: fermatas are basically the only thing that can go on a barline. So I would go to Barline::acceptDrop and remove ARTICULATION from the accepted items. This should prevent the problem from occuring again. I would also ask maybe @DmitryArefiev or @zacjansheski if they could do some more testing around this type of situation (dropping items on barlines) to see that the problem doesn't involve other stuff too :)